go/tendermint/apps/staking/state/state.go

```diff
@@ -543,6 +565,64 @@ func (s *MutableState) TransferFromCommon(ctx *abci.Context, toID signature.Publ
 	return ret, nil
 }
 
+// AddRewards computes and transfers the staking rewards to active escrow accounts.
+// If an error occurs, the pool and affected accounts are left in an invalid state.
+func (s *MutableState) AddRewards(time epochtime.EpochTime, factor *staking.Quantity, accounts map[signature.MapKey]bool) error {
+	steps, err := s.RewardSchedule()
+	if err != nil {
+		return err
+	}
+	var activeStep *staking.RewardStep
+	for _, step := range steps {
+		if time < step.Until {
+			activeStep = &step
+			break
+		}
+	}
+	if activeStep == nil {
+		// We're past the end of the schedule.
+		return nil
+	}
+
+	commonPool, err := s.CommonPool()
+	if err != nil {
+		return errors.Wrap(err, "loading common pool")
+	}
+
+	for mk := range accounts {
```

Comment:
Iteration over Go maps is non-deterministic and may cause consensus failures (different validators will produce different hashes due to IAVL being sensitive to order of updates). One idea is to add updates to a list and sort them before committing -- this would also solve the issue described below about possibly committing partial updates.

Reply:
I'll sort these